### ISSUE TITLE
generator beaks files when it crashes

### ISSUE THREAD

**Issue by nielsdrost** [NLeSC/software.esciencecenter.nl#93]
If the generator crashes (especially during the writing of files) it leaves these files in a "broken" state.

**Comment by blootsvoets** [comment]
Ah, you're right. It would be nicer to write to a temporary file, catch any errors, and only move the written file if no errors were raised.

**Comment by nielsdrost** [comment]
Perhaps create the content as a string? Moving temp files around can also go wrong ;-)

**Comment by blootsvoets** [comment]
yeah... opening a file to write and then realising you can't write the string also doesn't help though.
